Add examples how to integrate this in editors (#4)
remarkjs/remark-language-server: readme.md

Usage of this package depends on your editor integration.
Because this is based on
[`unified-languageserver-node`][unified-languageserver-node], the same features
are supported.

`remark-language-server` uses the same configuration files as
[`remark-cli`][remark-cli].

## Examples

### Atom

The [`linter-remark`][linter-remark] plugin for Atom uses remark language server.
It can be installed with:

```sh
apm install linter-remark

> ```

Reviewer: As this is pending, wouldn’t it be better to link to the pull request?
And, will Atom ever work?

Author: I don’t want to document this temporary situation where there’s an open pull request, but it hasn’t been released yet.

Depending on your stance on whether or not `linter-remark` should be updated, I think this should either stay the way it is (and be merged after https://github.com/wooorm/linter-remark/pull/15), or be removed.

### Emacs

> The remark language server has been registerd with [Emacs `lsp-mode`][lsp-mode].

Reviewer: - This phrasing is too weak: https://github.com/retextjs/retext-passive.
- `registerd` is a typo
- similar to `Vim-lsp`, should this explain how to use it?

```suggestion
Use [`lsp-mode`][lsp-mode] to use the remark language server with Emacs.
```

Author: > * similar to `Vim-lsp`, should this explain how to use it?

Preferably yes. I registered `remark-language-server`, but I’m not familiar with Emacs myself. I welcome any Emacs users to enhance this.

### Neovim

The remark language server has been registerd with [`nvim-lspconfig`][nvim-lspconfig].

> This means it’s available by default in Neovim.

Author: I’m guessing @Murderlon may be able to improve instructions for Neovim.

Reviewer: - This phrasing is too weak: https://github.com/retextjs/retext-passive.
- `registerd` is a typo
- should probably also explain how to use it

> This means it’s available by default in Neovim.

Reviewer: ```suggestion
Use [`nvim-lspconfig`][nvim-lspconfig] to use the remark language server with
Neovim.
It’s enabled by default.
```

> ### Vim-lsp

Reviewer: ```suggestion
### vim-lsp
```

`vim-lsp` is cased lowercase, even at sentence start: https://github.com/prabirshrestha/vim-lsp#snippets.

Reviewer: `vim-lsp` also works with neovim (previous section).
Perhaps both should be under a heading called `Vim` or `Vim / Neovim`

Reviewer: should it be documented why nvim-lspconfig or vim-lsp should be chosen over the other?

Author: These are preferences. The goal is just to list some integrations, so users of various editors can get started. It’s not to judge which integration should be chosen.

The remark language server can be used with [`vim-lsp`][vim-lsp].
You can configure it to be started when a markdown file is opened.

```viml
if (executable('remark-language-server'))
  au User lsp_setup call lsp#register_server({
  \ 'name': 'remark',
  \ 'cmd': {server_info->['remark-language-server', '--stdio']},
  \ 'allowlist': ['markdown'],
  \ })
endif
```

It has also been registered with [`vim-lsp-settings`][vim-lsp-settings].
If you use this, you will be prompted to use the remark language server when a
markdown file is opened.

### Visual Studio Code

> The [remark extension for Visual Studio Code][vscode-remark] uses remark
> language server under the hood.

Reviewer: ```suggestion
Use [`vscode-remark`][vscode-remark] to use the remark language server with
Visual Studio Code.
```

Usage of this package depends on your editor integration.
Because this is based on
[`unified-languageserver-node`][unified-languageserver-node], the same features
are supported.

`remark-language-server` uses the same configuration files as
[`remark-cli`][remark-cli].

## Examples

### Atom

The [`linter-remark`][linter-remark] plugin for Atom uses remark language server.
It can be installed with:

```sh
apm install linter-remark
```

### Emacs

Use [`lsp-mode`][lsp-mode] to use the remark language server with Emacs.

### Neovim

> Use [`nvim-lspconfig`][nvim-lspconfig] to use the remark language server with
> Neovim.
> It’s enabled by default.

Reviewer 2: ````suggestion
The remark language server is part of [`nvim-lspconfig`][nvim-lspconfig], a collection of common configurations for Neovim's built-in language server client. This means you can add remark language server in one line:

```lua
require'lspconfig'.remark_ls.setup{}
```

Detailed installation instructions can be found [here](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#remark_ls).
````

Reviewer 3: if it's possible to fallback to a global or bundled version of remark, can that be documented as well?

https://github.com/unifiedjs/unified-language-server/issues/31

Usage of this package depends on your editor integration.
Because this is based on
[`unified-languageserver-node`][unified-languageserver-node], the same features
are supported.

`remark-language-server` uses the same configuration files as
[`remark-cli`][remark-cli].

## Examples

### Emacs

Use [`lsp-mode`][lsp-mode] to use the remark language server with Emacs.

### Neovim

The remark language server is part of [`nvim-lspconfig`][nvim-lspconfig], a
collection of common configurations for Neovim’s built-in language server
client.
This means you can add remark language server in one line:

```lua
require'lspconfig'.remark_ls.setup{}
```

> Detailed installation instructions can be found [here](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#remark_ls).

Reviewer: `here` is not very accessible (screenreaders allow jumping between links and then they read the label, so “here”). Also, maybe use a reference/definition?

Reviewer 2: ```suggestion
`nvim-lspconfig` has [detailed installation instructions](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#remark_ls) such as configuration options.
```

### vim-lsp

The remark language server can be used with [`vim-lsp`][vim-lsp].
You can configure it to be started when a markdown file is opened.

```viml
if (executable('remark-language-server'))
  au User lsp_setup call lsp#register_server({
  \ 'name': 'remark',
  \ 'cmd': {server_info->['remark-language-server', '--stdio']},
  \ 'allowlist': ['markdown'],
  \ })
endif
```

> It has also been registered with [`vim-lsp-settings`][vim-lsp-settings].
> If you use this, you will be prompted to use the remark language server when a
> markdown file is opened.

Reviewer: ```suggestion
Otherwise, if you use [`vim-lsp-settings`][vim-lsp-settings], you will be asked
to use the remark language server automatically when a markdown file is opened.
```

Maybe like this? Less use of “this” and weaker language such as “has also been”
